tests/unit/igraph_circulant.c

```diff
@@ -0,0 +1,28 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+
+#include "test_utilities.inc"
+
+int main() {
+
+    // TODO: Add a test
+
+    return 0;
+}
```

Comment:
You can set your editor to always put a newline at the end of a file, that prevents this issue ( no newline at end of file). It's a really minor thing, but if you for example want to view a lot of files in one go by using `cat` then this matters.

Reply:
Or, just install [`pre-commit`](https://pre-commit.com/) and run `pre-commit install` in your clone of the repo; this will install our preferred set of pre-commit hooks that get executed before every commit. The pre-commit hooks will take care of trailing newlines for you.

Reply:
As you see, I'm also still learning things :D

Reply:
Ah I was wondering what the failed test was. Thanks!